### DISCUSSION TITLE
depth map/RGBD viewer for VR headsets?

### DISCUSSION THREAD

**Discussion by nigh8w0lf** [thygate/stable-diffusion-webui-depthmap-script#5] [Q&A]
Anyone know how to view these images in a VR headset with existing apps, especially for Gearvr?

**Comment by thygate** [comment]
I am not aware of any specific (android) apps for GearVR, but I have done this in the past on several other VR headsets using Unity3D and SteamVR.

Using a plane with a displacement modifier to render the image on, and then have 2 camera's offset horizontally render the output for the VR display.

You can do pretty much the same thing in any 3D game engine, or 3D graphics software like blender etc..

I will add a demo in blender to the repo later .

I also have a web based viewer using threejs that might be up to the task, not sure. I'll look into it, and might release a html based viewer later too.

Thanks for commenting.

**Reply by nigh8w0lf** [reply]
Thanks for the reply, I will dig around a bit more to see if I can find any, one potential option I found so far is VrPlayer app, haven't tested it though, saw some posts on reddit saying it might be rgbd capable. 

Look forward to the html threejs viewer if you have the time for it, thanks for creating this project!

**Reply by thygate** [reply]
Does your android browser & GearVR support WebXR ?

You can test it here :
https://webvr.rocks/samsung_gear_vr

**Comment by finefin** [comment]
I own a Quest 1 and the best way I found to view media is SteamVR+VirtualDesktop. I think VirtualDesktop is available for GearVR as well.

**Comment by keenerb** [comment]
There is a Pig media viewer for Quest that does a good job handling VR media, including old VR180 files.

https://www.meta.com/experiences/2436667223120459/

**Comment by hoagiemc** [comment]
On Meta Quests I use Skybox Video Player. You can view still images with it as well as videos.

**Comment by sonakravtsenko3** [comment]
This works for me:
First create stereoimages
Then use Sunshine streaming server + Moonlight for Android to view them
You can change desktop resolution to match aspect ratio of your phone.

You may also check out things like VRidge and Virtual Desktop.
